src/MainWindow.cpp

```diff
@@ -72,10 +76,16 @@ MainWindow::MainWindow(BaseObjectType* cobject, Glib::RefPtr<Gtk::Builder> const
     closeToTraySwitch->set_state(m_trayIcon.isVisible());
     startInTraySwitch->set_state(Settings::getInstance().getStartInTray() && m_trayIcon.isVisible());
     startInTraySwitch->set_sensitive(m_trayIcon.isVisible());
+    autostartSwitch->set_state(Settings::getInstance().getAutostart());
 
     m_headerBar->set_visible(Settings::getInstance().getHeaderBar());
 }
 
+MainWindow::~MainWindow()
+{
+    Settings::getInstance().autostart();
```

Comment:
Please rename the function as `setAutoStart`

Reply:
I removed this completely. With moving autostart to setAutostart,  We do not need dtor. This was needed when performing autostart only once.